test/unit/change_trust_asset_test.js

```diff
@@ -0,0 +1,355 @@
+
+    it('returns "credit_alphanum12" if the trustline asset code length is between 5 and 12', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'ABCDEF',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      expect(asset.getAssetType()).to.eq('credit_alphanum12');
+    });
+
+    it('returns "liquidity_pool_shares" if the trustline asset is a liquidity pool ID', function() {
+      const asset = StellarBase.ChangeTrustAsset.liquidityPoolAsset({
+        assetA,
+        assetB,
+        fee
+      });
+      expect(asset.getAssetType()).to.eq('liquidity_pool_shares');
+    });
+  });
+
+  describe('toXDRObject()', function() {
+    it('parses a native asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.native();
+      const xdr = asset.toXDRObject();
+      expect(xdr.toXDR().toString()).to.be.equal(
+        Buffer.from([0, 0, 0, 0]).toString()
+      );
+    });
+
+    it('parses a 3-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'USD',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum4');
+      expect(xdr.value().assetCode()).to.equal('USD\0');
+    });
+
+    it('parses a 4-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        'BART',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum4');
+      expect(xdr.value().assetCode()).to.equal('BART');
+    });
+
+    it('parses a 5-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        '12345',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum12');
+      expect(xdr.value().assetCode()).to.equal('12345\0\0\0\0\0\0\0');
+    });
+
+    it('parses a 12-alphanum asset object', function() {
+      const asset = StellarBase.ChangeTrustAsset.creditAsset(
+        '123456789012',
+        'GCEZWKCA5VLDNRLN3RPRJMRZOX3Z6G5CHCGSNFHEYVXM3XOJMDS674JZ'
+      );
+      const xdr = asset.toXDRObject();
+
+      expect(xdr).to.be.instanceof(StellarBase.xdr.ChangeTrustAsset);
+      expect(() => xdr.toXDR('hex')).to.not.throw();
+
+      expect(xdr.arm()).to.equal('alphaNum12');
+      expect(xdr.value().assetCode()).to.equal('123456789012');
```

Comment:
I don't like how there are 4 near-identical tests here; can we consolidate them somehow, e.g. into a loop?

Reply:
When we're talking about tests I prefer being verbose and have more readability. Using a for loop here would save us 2 or 3 lines per test at the cost of reduced readability so I don't think it's worth it for this particular case. WDYT?